src/components/SingletonEditor.tsx

```diff
@@ -56,11 +62,17 @@ export const useSingletonEditor = ({
 
       onSave(content as Content);
     });
-  };
+  }, [editorState, onSave]);
+
+  useEffect(() => {
+    window.removeEventListener("beforeunload", saveHandler);
+    window.addEventListener("beforeunload", saveHandler);
+  }, [saveHandler]);
 
   useEffect(() => {
     setContent(initialContent);
     return () => {
+      saveHandler();
```

Comment:
Oh, it is a different feature: it ensures to save on component unmount. What do you think about use it in a different commit than `beforeunload` related save?

Reply:
We talked with @Cromeror, and we agree in that these are two different features, but it is not necessary to split into different commits.